### ISSUE TITLE
(aws-cloudfront): Lambda at Edge Cloudwatch log groups retention

### ISSUE THREAD

**Issue by izifortune** [aws/aws-cdk#12690]
When creating a new lambda at edge, the retention policy is ignored for all the regional loggroups created causing the loggroups to never expire

### Reproduction Steps

1. Create a lambda function:

```
    const originRequestFunction = new Function(this, 'originRequestFunction', {
        ...
      logRetention: RetentionDays.TWO_WEEKS,
    });
```

2. Associate the lambda function to cloudfront:

```
            lambdaFunctionAssociations: [
              {
                eventType: LambdaEdgeEventType.ORIGIN_REQUEST,
                lambdaFunction: Version.fromVersionArn(
                  this,
                ....
              }
            ]
```

### What did you expect to happen?

The regional loggroups (regional-cache) should follow the retention period specified in the lambda function definition

### What actually happened?

The regional loggroups don't have any expiration

### Environment

  - **CDK CLI Version  :** 1.60
  - **Framework Version:** 1.60
  - **Node.js Version:** 12.x
  - **OS               :** AmazonLinux
  - **Language (Version):** TypeScript 

--- 

This is :bug: Bug Report

**Comment by njlynch** [comment]
This is a known issue with the way Lambda@Edge log groups are handled. See https://github.com/serverless/serverless/issues/7574 for another discussion on this topic.

It appears the solution here would be to either: (a) proactively create the log groups for every possible region in advance, with the retention period set; (b) create some kind of job that periodically checks for these log groups and updates the retention period; or (c) wait for the Lambda team to change how this is processed (see https://forums.aws.amazon.com/message.jspa?messageID=899555 for a thread from May 2019 where the feedback was noted). Of the above, option B seems potentially the best balance, but quite the hack.

I'm going to rebrand this as a feature request, as the "home" buckets' retention policies are being honored.

**Comment by izifortune** [comment]
Thanks @njlynch I’ve been following also the serverless one as I have some lambdas with the framework as well.

Regarding the options that you listed:

a) will the regions depend on the price class of the distribution? That would probably add extra complexity.

b) seems a viable solution but probably will need to be explored outside of the cdk?

**Comment by njlynch** [comment]
Yes, I believe the regions would depend on the price class. Option B could be done within the CDK, but also equally from outside the CDK as a separate construct. I'd certainly be willing to look at draft PRs for a contribution to the CDK.

**Comment by 70nyIT** [comment]
any updates on this issue? I'm deploying lambda @ edge and I'd like to make sure we keep the size of these logs under control.

**Comment by Kevinwochan** [comment]
https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_cloudfront.experimental.EdgeFunction.html#logretention

Is currently in experimental mode

**Comment by doteric** [comment]
> https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_cloudfront.experimental.EdgeFunction.html#logretention
> 
> Is currently in experimental mode

Hey @Kevinwochan  
On the mentioned link there is:

> This is a legacy API and we strongly recommend you move away from it if you can. Instead create a fully customizable log group with logs.LogGroup and use the logGroup property to instruct the Lambda function to send logs to it. Migrating from logRetention to logGroup will cause the name of the log group to change. Users and code and referencing the name verbatim will have to adjust.

We create a Log Group for our AWS Lambdas@Edge ([via terraform](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/lambda_function)). The initially created CW Log Group is fine, the problem is with all the edge log groups that are automatically created by AWS. I do not think this a problem of the API, **I think AWS should apply the same retention period to created edge lambda log groups as on the main CloudWatch log group**. Instead it's always set to never expire and needs to be manually set AFTER it's creation, in ALL regions and there is no real other way..  
This has been reported to AWS Support and marked as a feature in the future, but we have no further visibility or information on this.
